tests/integration/server/roles_api_test.py

```diff
@@ -94,3 +99,207 @@ def test_syncdirectionflag_rolename():
+            "description": "Test Role 1",
+        }
+        role_response = create_test_role(session, role_payload)
+        role_result = assert_api_success(role_response)
+        role_id = role_result["data"]["id"]
+        role_update_payload = {
+            "id": user2_id,
+            "reason": "Integration test of adding role admin.",
+            "metadata": "",
+        }
+        response = session.post(
+            "http://rbac-server:8000/api/roles/{}/admins".format(role_id),
+            json=role_update_payload,
+        )
+        result = assert_api_success(response)
+        proposal_response = get_proposal_with_retry(session, result["proposal_id"])
+        proposal = assert_api_success(proposal_response)
+        assert proposal["data"]["assigned_approver"][0] == user1_id
+        delete_role_by_name("TestRole1")
+        delete_user_by_username("testuser1")
+        delete_user_by_username("testuser2")
+
+
+def test_add_role_owner():
+    """Test adding an owner to a role.
+
+    Creates two test users and a role with user1 as owner/admin,
+    then adds the second user as role owner."""
+    user1_payload = {
+        "name": "Test User 3",
+        "username": "testuser3",
+        "password": "123456",
+        "email": "testuser3@biz.co",
+    }
+    user2_payload = {
+        "name": "Test User 4",
+        "username": "testuser4",
+        "password": "123456",
+        "email": "testuser4@biz.co",
+    }
+    with requests.Session() as session:
+        user_response1 = create_test_user(session, user1_payload)
+        user1_result = assert_api_success(user_response1)
+        user1_id = user1_result["data"]["user"]["id"]
+        user2_response = create_test_user(session, user2_payload)
+        user2_result = assert_api_success(user2_response)
+        user2_id = user2_result["data"]["user"]["id"]
+        role_payload = {
+            "name": "TestRole2",
+            "owners": user1_id,
+            "administrators": user1_id,
+            "description": "Test Role 2",
+        }
+        role_response = create_test_role(session, role_payload)
+        role_result = assert_api_success(role_response)
+        role_id = role_result["data"]["id"]
+        role_update_payload = {
+            "id": user2_id,
+            "reason": "Integration test of adding role owner.",
+            "metadata": "",
+        }
+        response = session.post(
+            "http://rbac-server:8000/api/roles/{}/owners".format(role_id),
+            json=role_update_payload,
+        )
+        result = assert_api_success(response)
+        proposal_response = get_proposal_with_retry(session, result["proposal_id"])
+        proposal = assert_api_success(proposal_response)
+        assert proposal["data"]["assigned_approver"][0] == user1_id
+        delete_role_by_name("TestRole2")
+        delete_user_by_username("testuser3")
+        delete_user_by_username("testuser4")
+
+
+def test_add_role_member():
```

Comment:
Since the scope of this ticket is limited to the created proposal it makes sense to only check for the assigned_approvers field in the proposal, but it might be worth approving the proposal and checking for the change in rethink (or just adding some `TODO`'s for this). @nadiabahrami or @mtn217 can weigh in, this won't block approval if it'll take too long to add.

Reply:
I agree, but it'll take "too long" in the sense that I have no available time to do that until next week.